x-pack/elastic-agent/pkg/agent/operation/monitoring.go

```diff
@@ -324,6 +326,64 @@ func (o *Operator) getMonitoringMetricbeatConfig(output interface{}) (map[string
 			},
 		})
 	}
+
+	// setup cpu, memory and fd monitors for agents
+	modules = append(modules, map[string]interface{}{
+		"module":     "system",
+		"period":     "10s",
+		"metricsets": []string{"process"},
+		"index":      fmt.Sprintf("metrics-elastic_agent.%s-default", agentName),
+		"processes":  []string{install.BinaryName},
+		"processors": []map[string]interface{}{
+			{
+				"add_fields": map[string]interface{}{
+					"target": "data_stream",
+					"fields": map[string]interface{}{
+						"type":      "metrics",
+						"dataset":   fmt.Sprintf("elastic_agent.%s", agentName),
```

Comment:
Do we do this for logs? Is it `elastic_agent.elastic-agent`? I think its just `elastic_agent`.

The `-` in `elastic-agent` is a problem as well. We should not have that, because we had to change `endpoint-security` to `endpoint_security` otherwise it breaks how the namespace is used of ending with `-default`.